src/components/todolist/TodoList.js

```diff
@@ -1,17 +1,31 @@
-import React from 'react';
-import { useSelector } from 'react-redux'
-import TodoItems from '../todoitems/TodoItems';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { completionStatus, moveToTrash } from '../../actions/todoActions';
+import { TodoItems } from '../todoitems/TodoItems';
 import './todolist.css'
 
 export const TodoList = () => {
     const todoData = useSelector(state => state)
-    
+    let dispatch = useDispatch()
+
+    const checkHandler = (p) => {
+        dispatch(completionStatus(p.data.id))
+    }
+
+    const trashhandler = (p) => {
+        dispatch(moveToTrash(p.data.id))
+    }
+
     return (
         <div className='mainContainer'>
             {
                 (todoData !== undefined)
                     ? todoData.map(data => (
-                        (!data.trashed) ? <TodoItems key={data.id} data={data} /> : <></>
+                        (!data.trashed) ? <TodoItems key={data.id}
+                            data={data}
+                            onCheckChange={(p) => { checkHandler(p) }}
```

Comment:
this will also work but we can also replace it with onCheckChange={onCheckChange}

Reply:
done